src/Operations.jl

```diff
@@ -1033,10 +1034,12 @@ function update_package_pin!(pkg::PackageSpec, ::Nothing)
     if pkg.version == VersionSpec() # no version to pin
         pkgerror("Can not `pin` a package which does not exist in the manifest")
     end
+    is_stdlib(pkg.uuid) && pkgerror("cannot `pin` stdlibs.")
     pkg.pinned = true
 end
 
 function update_package_pin!(pkg::PackageSpec, entry::PackageEntry)
+    is_stdlib(pkg.uuid) && pkgerror("cannot `pin` stdlibs.")
```

Comment:
Probably does not make sense to `pin` a stdlib. It is an error in 1.0 (`ERROR: KeyError: key "version" not found`) and 1.1 (`ERROR: MethodError: no method matching VersionNumber(::Nothing)`) so this is not a regression, and results in a better error message.